inc/infocom.class.php

```diff
@@ -2023,4 +2023,27 @@ function canCreateItem() {
       return Session::haveRight(static::$rightname, CREATE);
    }
 
+   /**
+    * Get item types
+    *
+    * @since 9.3.1
+    *
+    * @param array $where Where clause
+    *
+    * @return DBMysqlIterator
+    */
+   public static function getTypes($where) {
+      global $DB;
+
+      $types_iterator = $DB->request([
+         'SELECT DISTINCT' => 'itemtype',
+         'FROM'            => 'glpi_infocoms',
+         'WHERE'           => [
+            'NOT'          => ['itemtype' => ['ConsumableItem', 'CartridgeItem', 'Software']]
```

Comment:
it could be better to keep ignore list in a separate var, to have better readability